src/app/selectors.js

```diff
@@ -99,6 +99,20 @@ const getFilteredModelData = (filters) =>
 
 // ---- Utility functions
 
+/**
+  Some models can be namespaced by user so we must validate the url params
+  before confirming corrrect model name
+*/
+export const validateModelNameFromURL = (param1, param2) => {
```

Comment:
Instead of doing this can we just make the username required as part of the url?

IIRC there is only a single place where we generate links in the application to models and that's in the model list tables. So by modifying [generateModelDetailsLink](https://github.com/canonical-web-and-design/jaas-dashboard/blob/master/src/components/ModelTableList/shared.js#L13) by removing the first conditional it'll always return links with the usernames in them. In doing so this function isn't necessary.